src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/main.rs

```diff
@@ -122,15 +91,8 @@ impl BlockNumberResult {
 }
 
 #[marine]
-pub fn latest_block_number(net: String) -> BlockNumberResult {
-    let url = match get_url(&net) {
-        None => {
-            return BlockNumberResult::error(Error::NetworkTypeError(net).to_string());
-        }
-        Some(url) => url,
-    };
-
-    let result = match get_block_number(url) {
+pub fn latest_block_number(api_endpoint: String) -> BlockNumberResult {
```

Comment:
Maybe parse it to URI instead of having as a string? That would catch incorrect URI earlier.